### DISCUSSION TITLE
binding for qml and flutter

### DISCUSSION THREAD

**Discussion by rraammiinn** [vlang/v#17243] [Questions and Answers]
will be any binding for qml or flutter ?

**Comment by JalonSolov** [comment]
That sounds like a good project for someone as an external module.

**Comment by trufae** [comment]
I dont think you can use flutter from anywhere else than Dart. And qml is built on top of c++ and qt, so its inherently not usable from C. So, considering both projects are privative and depending on huge and moving frameworks i dont see it reasonable to be achieved and in any case it should be done outside the core v repo

**Reply by rraammiinn** [reply]
qml has binding for many languages so why not v ?

**Reply by rraammiinn** [reply]
https://wiki.qt.io/Language_Bindings

**Reply by trufae** [reply]
There's no C in that list. You'll need to get V to generate c++ code instead for this to be made. Its possible to generate c bindings for c++ apis like wxwidgets do. But afaik this is not possible for all the c++ constructions and im unsure if that will work for Qt

**Reply by Wajinn** [reply]
Arguably, something more feasible than qt, would be raylib and add raygui too.  There are already V raylib bindings out there, and Go examples of bindings for other headers from them.

**Reply by Wajinn** [reply]
> qml has binding for many languages so why not v ?

> https://wiki.qt.io/Language_Bindings

Qt has licensing that can be very unattractive to various people.  Also, V has integrated sokol very well.  There is VUI, MUI, and Isaiah's UI.  And which are under the friendlier MIT or Apache.

**Reply by rraammiinn** [reply]
what about flutter ?
